x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/optimizer/LocalPhysicalPlanOptimizerTests.java

```diff
@@ -139,7 +140,7 @@ public void init() {
 
     private Analyzer makeAnalyzer(String mappingFileName, EnrichResolution enrichResolution) {
         var mapping = loadMapping(mappingFileName);
-        EsIndex test = new EsIndex("test", mapping);
+        EsIndex test = new EsIndex("test", mapping, Set.of("test"));
```

Comment:
With all the similar changes to add a concrete index, wondering if `EsIndex(String name, Map<String, EsField> mapping)` c'tor, which is only used in tests, shouldn't be updated instead to add a set with the `name`.

Reply:
I have mixed feelings about this. I know it's not a big deal, but if something is used only in tests, the method/constructor access modifier should reflect this. But this class is used in many different packages and the only possible access modifier is `public`. I would rather have the usage be explicit and the user of the constructor be aware of the implications of providing an empty Set or specific values in that Set, rather than use the constructor without being aware of its content.